client/consensus/slots/src/lib.rs

```diff
@@ -644,4 +732,209 @@ mod test {
 			Some(SLOT_DURATION * 2u32.pow(7)),
 		);
 	}
+
+	#[derive(PartialEq, Debug)]
+	struct HeadState {
+		head_number: NumberFor<Block>,
+		head_slot: u64,
+		slot_now: NumberFor<Block>,
+	}
+
+	impl HeadState {
+		fn author_block(&mut self) {
+			// Add a block to the head, and set latest slot to the current
+			self.head_number += 1;
+			self.head_slot = self.slot_now;
+			// Advance slot to next
+			self.slot_now += 1;
+		}
+
+		fn dont_author_block(&mut self) {
+			self.slot_now += 1;
+		}
+	}
+
+	#[test]
+	fn should_never_backoff_when_head_not_advancing() {
+		let strategy = SimpleBackoffAuthoringBlocksStrategy::<NumberFor<Block>> {
+			max_interval: 100,
+			unfinalized_slack: 5,
+			authoring_bias: 2,
+		};
+
+		let head_number = 1;
+		let head_slot = 1;
+		let finalized_number = 1;
+		let slot_now = 2;
+
+		let should_backoff: Vec<bool> = (slot_now..100)
```

Comment:
Wouldn't this eventually back off if we did this from `(slot_now..103)`? In which case the reason this isn't backing off is because we didn't reach the interval and not because the head isn't advancing.

Reply:
I upped the range to `(slot_now...1000)` to check and it's still always false